### ISSUE TITLE
Implement 1D BatchNorm 

### ISSUE THREAD

**Issue by csuraparaju** [csuraparaju/dnn_rs#7]
## Description

Z-score normalization is the procedure during which the feature values are rescaled so that they have the properties of a normal distribution. Let $\mu$ be the mean (the average value of the feature, averaged over all examples in the dataset) and $\sigma$ be the standard deviation from the mean. The Z-scores of features are calculated as follows:
$$\hat{a} =   \frac{x - \mu}{\sigma}$$

Batch normalization is a method used to make training of artificial neural networks faster and more stable through normalization of the layers’ inputs by re-centering and re-scaling.

## Proposed Solution

1. Model a 1D `BatchNorm` trait and define a struct that implements it. 
2. Use the calculus described [here](https://deeplearning.cs.cmu.edu/F23/document/homework/HW1/HW1P1_F23.pdf) to derive the necessary forward and backward functions for a 1D `BatchNorm` layer.
